src/signalcorr.jl

```diff
@@ -513,7 +513,7 @@ function pacf_regress!(r::RealMatrix, X::AbstractMatrix{T}, lags::IntegerVector,
         for i = 1 : length(lags)
             l = lags[i]
             sX = view(tmpX, 1+l:lx, 1:l+1)
-            r[i,j] = (cholfact!(sX'sX)\(sX'view(X, 1+l:lx, j)))[end]
+            r[i,j] = l == 0 ? one(T) : (cholfact!(sX'sX)\(sX'view(X, 1+l:lx, j)))[end]
```

Comment:
Will the result of `(cholfact!(sX'sX)\(sX'view(X, 1+l:lx, j)))[end]` definitely have type `T`?

Reply:
The way I understand it, `T <:RealFP` (where `RealFP = Union{Float32, Float64}` - defined in common.jl, line 21) means that the matrix `X` and therefore also `sX` is a real floating point matrix. That would also imply that the correlation coefficient is a floating point number (Float32 if `X` is Float32 and Float64 if `X` is Float64).

This is also how it is handled in `pacf_yulewalker!()` (same file, line 528). Do you have a better way? I'm happy to change it.

Reply:
AFAICT `one(T)` isn't better than `1` anyway since implicit conversion will happen.

Reply:
Ok. I replaced `one(T)` by `1`. Should we then also change it in `pacf_yulewalker!()` to be consistent?

Reply:
No idea, I don't know that code.

Reply:
Ok, thanks. Maybe @andreasnoack can help here.

Reply:
Yes, I think it would be good to change it there for consistency.

Reply:
Done! Thanks for the quick feedback.